crates/services/p2p/src/codecs/postcard.rs

```diff
@@ -161,29 +177,214 @@ impl GossipsubCodec for PostcardCodec {
 }
 
 impl NetworkCodec for PostcardCodec {
-    fn get_req_res_protocol(&self) -> <Self as request_response::Codec>::Protocol {
-        MessageExchangePostcardProtocol {}
+    fn get_req_res_protocols(
+        &self,
+    ) -> impl Iterator<Item = <Self as request_response::Codec>::Protocol> {
+        // TODO: Iterating over versions in reverse order should prefer
+        // peers to use V2 over V1 for exchanging messages. However, this is
+        // not guaranteed by the specs for the `request_response` protocol.
+        PostcardProtocol::iter().rev()
     }
 }
 
-#[derive(Default, Debug, Clone)]
-pub struct MessageExchangePostcardProtocol;
+#[derive(Debug, Default, Clone, EnumIter)]
+pub enum PostcardProtocol {
+    #[default]
+    V1,
+    V2,
+}
```

Comment:
Why `V1` is default if we prefer `V2`?

Reply:
Thanks, this is a leftover for when I was doing protocol selection manually. In fact, we don't even need a Default version. 
I have removed it in the follow-up MR: https://github.com/FuelLabs/fuel-core/pull/2377 (commit ea4842c3c6)